File: codility/src/main/java/codility/com/lessons/l05prefixsums/MinAvgTwoSlice.java
package codility.com.lessons.l05prefixsums;


/**
 * 
 * @author 08115341622
 */
public class MinAvgTwoSlice {
	public int solution(int[] A) {
		double minAvg = 100000;
		int index = 0;

		if (A.length <= 2) {
			return 0;
		}

		for (int i = 0; i < A.length - 2; i++) {
			if ((A[i] + A[i + 1]) / 2.0 < minAvg) {
				minAvg = (A[i] + A[i + 1]) / 2.0;
				index = i;
			}

			if ((A[i] + A[i + 1] + A[i + 2]) / 3.0 < minAvg) {
				minAvg = (A[i] + A[i + 1] + A[i + 2]) / 3.0;
				index = i;
			}
		}

		int aMax = A.length - 2;
		if ((A[aMax] + A[aMax + 1]) / 2.0 < minAvg) {
			minAvg = (A[aMax] + A[aMax + 1]) / 2.0;
			index = aMax;
		}

		return index;
	}
}

File: codility_new/min_avg_two_slice.py
# https://app.codility.com/programmers/lessons/5-prefix_sums/min_avg_two_slice/
def solution(A):
    min_found = 100_000
    start_pos = 0
    prefix_sum = [0] * (len(A) + 1)
    for i in range(len(A)):
        prefix_sum[i + 1] = prefix_sum[i] + A[i]
    
    # print(f"prefix_sum: {prefix_sum}")
    for p in range(len(A)):
        for q in range(p + 1, min(len(A), p + 3)):
            # print(f"p: {p}, q: {q}")
            slice_value = (prefix_sum[q + 1] - prefix_sum[p]) / (q - p + 1)
            # print(f"p: {p}, q: {q},  slice_value: {slice_value}")
            if slice_value < min_found:
                # print(f"p: {p}, q: {q}, min_found: {min_found}, slice_value: {slice_value}, start_pos: {start_pos}")
                min_found = slice_value
                start_pos = p
    
    return start_pos
